prover/src/lib.rs

```diff
@@ -63,11 +63,30 @@ where
     )?;
     #[cfg(feature = "std")]
     debug!(
-        "Generated execution trace of {} columns and {} steps in {} ms",
+        "Generated execution trace of {} columns in {} ms",
         trace.layout().main_trace_width(),
-        trace.length(),
         now.elapsed().as_millis()
     );
+    #[cfg(feature = "std")]
+    debug!(
+        "VM cycles: {} extended to {} steps.
+├── Stack rows: {}
+├── Range checker rows: {}
+└── Chiplets rows: {}
+    ├── Hash chiplet rows: {}
+    ├── Bitwise chiplet rows: {}
+    ├── Memory chiplet rows: {}
+    └── Kernel ROM rows: {}",
+        trace.trace_len_summary().trace_len(),
+        trace.trace_len_summary().padded_trace_len(),
+        trace.trace_len_summary().main_trace_len(),
+        trace.trace_len_summary().range_trace_len(),
+        trace.trace_len_summary().chiplets_trace_len().trace_len(),
+        trace.trace_len_summary().chiplets_trace_len().hash_chiplet_len(),
+        trace.trace_len_summary().chiplets_trace_len().bitwise_chiplet_len(),
+        trace.trace_len_summary().chiplets_trace_len().memory_chiplet_len(),
+        trace.trace_len_summary().chiplets_trace_len().kernel_rom_len(),
+    );
```

Comment:
Let's keep this largely as it was before. Maybe one thing I'd change is include the ratio of padding. For example:
```
Generated execution trace of 70 columns and 1048576 steps (20% padded) in 300 ms
```